### PR TITLE
Fix use of CSSTransitionGroup to work w/ modals.

### PR DESCRIPTION
**Pull request by toolness**
(This is a potential fix for PR https://github.com/mozilla/teach.webmaker.org/pull/712. It's fine if it doesn't get merged in--I just had to put my solution somewhere and pasting the diff into a comment was a bit large-ish.)

The `CSSTransitionGroup` is funky and a bit hard to understand.

In order to have a CSS transition occur, the `CSSTransitionGroup` needs to exist both _before_ and _after_ the content you want to transition exists in React's virtual DOM, or else it won't actually have time to transition the content.

In your existing code, the lifetime of the `CSSTransitionGroup` was identical to the lifetime of the modal, so there wasn't actually any time for it to transition the modal in or out. But if we move that `CSSTransitionGroup` out to `Page.jsx` and have it _always_ exist, and only give it children when there is actually a modal to be shown, then it will have all the time it needs to transition its children in and out, since it's always in React's virtual DOM.

(Sorry, I think this PR actually contains stuff in `develop` that isn't yet in your feature branch, which makes this PR's diff log a bit hard to parse. The only useful commit in this PR is https://github.com/toolness/Mozilla-Learning/commit/43022b1ca5412d07cfee5bfb8a985a7ddd150e94 .)

### CODE REVIEW

File: components/page.jsx [hunk 1]
```diff
@@ -75,13 +76,16 @@ var Page = React.createClass({
           </div>
           <Footer/>
         </div>
+
+        <ReactCSSTransitionGroup transitionName="modal">
         {this.state.modalClass
-         ? <div ref="modalHolder">
+         ? <div ref="modalHolder" key={1}>
```
Comment (toolness) [inline components/page.jsx:82]: We're just giving it a static key because there will only ever be one modal in this "list". I guess it may have the side effect of having no transition effect if we were to suddenly switch from one modal to another, but that doesn't currently happen in the codebase, and even if it did, no transition might actually be the behavior we want.

**Comment by toolness** [comment]
Oh, I should also note that the "enter" transition is a bit wonky on my Firefox--sometimes it happens, and sometimes it doesn't (the modal just instantly has 100% opacity). It seems to work consistently on Chrome, though, so my guess is that it's either a bug in `CSSTransitionGroup` or possibly even an oddity with the way Firefox does CSS transitions? I have no idea.

**Comment by mmmavis** [comment]
Thanks for the PR and explanation, @toolness :muscle: 

> In order to have a CSS transition occur, the CSSTransitionGroup needs to exist both before and after the content you want to transition exists in React's virtual DOM, or else it won't actually have time to transition the content.

This is something that confused me.  I know `CSSTransitionGroup` needs to exist both before and after the child component but the [code snippet #1 on React docs](https://facebook.github.io/react/docs/animation.html#getting-started) seems to be coded in a way that both `CSSTransitionGroup` and `{items}` have the same lifespan?  That snippet is where I mocked my approach from but my initial PR didn't work :sob: .  At first I thought it's cuz of the transition happening so fast that I couldn't catch it, that's why I changed the animation duration to 10s.  Um I still have trouble understanding why the code example would work but not my mocked approach.

But yea your PR totally makes sense for me, thanks!

**Comment by toolness** [comment]
Ah, you're right--I think [the one in the docs](https://facebook.github.io/react/docs/animation.html#getting-started) is actually a bad (or at least confusing) example. In that example, the initial items in the list--`['hello', 'world', 'click', 'me']`--_will_ actually appear instantaneously and not transition in. It's only when that example component's `handleAdd` or `handleRemove` methods are called that any transition effect will occur. The React docs should be more explicit about that!
